src/templates/virtual_services-ingress.ctmpl

```diff
@@ -19,11 +26,8 @@ spec:
   gateways:
     - default-gateway
   hosts:
-    - {{ $publicDomain }}
-    - "*.{{ $publicDomain }}"
-{{- if keyExists (print "app_config_vars/global/config.json/" $namespace "/SECONDARY_DOMAIN") }}
-    - {{ key (print "app_config_vars/global/config.json/" $namespace "/SECONDARY_DOMAIN") }}
-    - "*.{{ key (print "app_config_vars/global/config.json/" $namespace "/SECONDARY_DOMAIN") }}"
+{{- range $Key, $Value := scratch.Get (print $applicationName "-publicDomainList") }}
+    - "{{ $Key }}"
```

Comment:
I add the primary and secondary domain to a scratch map above, and then iterate over them here.
This means that if the same domain appears in more than one place it gets overwritten so it's only added here once.